packages/react-native/src/Authenticator/common/DefaultContent/types.ts

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { StyleProp, TextStyle, ViewStyle } from 'react-native';
+
+import {
+  AuthenticatorComponentDefaultProps,
+  AuthenticatorRouteComponentName,
+  AuthenticatorHeaderComponent,
+  AuthenticatorFooterComponent,
+} from '@aws-amplify/ui-react-core';
+
+import { ButtonProps } from '../../../primitives';
+
+import {
+  DefaultFormFieldsComponent,
+  DefaultFormFieldsStyle,
+} from '../../common';
+import { TextFieldOptionsType, RadioFieldOptions } from '../../hooks';
+
+export interface DefaultContentStyle {
+  buttonPrimary?: ViewStyle;
+  buttonPrimaryLabel?: TextStyle;
+  buttonSecondary?: ViewStyle;
+  buttonSecondaryContainer?: ViewStyle;
+  buttonSecondaryLabel?: TextStyle;
+  errorMessage?: ViewStyle;
+  errorMessageLabel?: TextStyle;
+  fieldContainerStyle?: ViewStyle;
+  fieldErrorStyle?: TextStyle;
+  fieldLabelStyle?: TextStyle;
+  fieldStyle?: TextStyle;
+  footer?: TextStyle;
+  formFields?: ViewStyle;
+  header?: TextStyle;
+}
+
+export type DefaultContentStyleProps = {
+  [Key in keyof DefaultContentStyle]: StyleProp<DefaultContentStyle[Key]>;
+} & DefaultFormFieldsStyle;
+
+type DefaultComponentProps<FieldsType> =
+  AuthenticatorComponentDefaultProps<FieldsType>[AuthenticatorRouteComponentName];
+
+type AuthenticatorButtonProps = Omit<ButtonProps, 'children'> & {
+  children: string;
+};
+
+type DefaultButtons = {
+  primary: AuthenticatorButtonProps;
+  secondary: AuthenticatorButtonProps | AuthenticatorButtonProps[];
+};
+
+export type DefaultContentProps<
+  FieldsType extends
+    | TextFieldOptionsType
+    | RadioFieldOptions
+    | unknown = unknown
+> = Pick<
+  DefaultComponentProps<FieldsType>,
+  'error' | 'Footer' | 'isPending'
+> & {
+  buttons: DefaultButtons;
+  body?: React.ReactNode;
```

Comment:
This needs to be retyped/refactored so it can handle `style` props for subcomponents that require unique content, for example `SetupTOTP`